wgpu-core/src/instance.rs

```diff
@@ -200,6 +208,9 @@ impl<B: hal::Backend> Adapter<B> {
                 .max(default_limits.max_uniform_buffers_per_shader_stage),
             max_uniform_buffer_binding_size: (adapter_limits.max_uniform_buffer_range as u32)
                 .max(default_limits.max_uniform_buffer_binding_size),
+            max_push_constant_size: (adapter_limits.max_push_constants_size as u32)
+                .max(default_limits.max_push_constant_size)
+                .max(MIN_PUSH_CONSTANT_SIZE),
```

Comment:
I'm curious if we need this `MIN_PUSH_CONSTANT_SIZE` at all, given that our `default_limits` should have exactly the same thing, and default limits reflect the baseline?

Reply:
For the case of extensions, the default is zero (as requesting non-zero requires the extension), but DX12 doesn't expose the push constant limit, so it's impossible to use push constants on DX12 right now unless we make set a minimum. All platforms should support 128, so until DX12 etc if fixed, this is what we need.

Reply:
Ok, I see. This is an interesting problem.
if we know that default_limits.max_push_constant_size` must be 0 because it's an extension, maybe we shouldn't do this `max(default_limits.max_push_constant_size)` here? (while we keep `.max(MIN_PUSH_CONSTANT_SIZE)`)

Reply:
Sure, that works.